React/Views/RCTWebViewManager.m

```diff
@@ -97,6 +97,18 @@ - (UIView *)view
   }];
 }
 
+RCT_EXPORT_METHOD(evaluateJavaScript:(nonnull NSNumber *)reactTag script: (nonnull NSString *)script callback:(RCTResponseSenderBlock)callback)
+{
+  [self.bridge.uiManager addUIBlock:^(__unused RCTUIManager *uiManager, NSDictionary<NSNumber *, RCTWebView *> *viewRegistry) {
+    RCTWebView *view = viewRegistry[reactTag];
+    if (![view isKindOfClass:[RCTWebView class]]) {
+      RCTLogError(@"Invalid view returned from registry, expecting RCTWebView, got: %@", view);
+    } else {
+      callback(@[[NSNull null], [view evaluateJavaScript: script]]);
```

Comment:
RN modules support promise natively, so changing this to use a promise instead of a callback will reduce boilerplate on the JS side.